shared/ghutil/pullrequest.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// pullrequest.go provides generic functions related to PullRequest
+
+package ghutil
+
+import (
+	"fmt"
+
+	"github.com/google/go-github/github"
+)
+
+const (
+	// PullRequestOpenState is the state of open PullRequest
+	PullRequestOpenState PullRequestStateEnum = "open"
+	// PullRequestCloseState is the state of closed PullRequest
+	PullRequestCloseState PullRequestStateEnum = "closed"
+	// PullRequestAllState is the state for all, useful when querying PullRequest
+	PullRequestAllState PullRequestStateEnum = "all"
+)
+
+// PullRequestStateEnum represents different states of PullRequest
+type PullRequestStateEnum string
+
+// ListPullRequests lists pull requests within given repo, filters by head user and branch name if
+// provided as "user:ref-name", and by base name if provided, i.e. "master"
+func (gc *GithubClient) ListPullRequests(org, repo, head, base string) ([]*github.PullRequest, error) {
+	PRsListOptions := github.PullRequestListOptions{
+		State: "all",
+	}
+	if "" != head {
```

Comment:
what happens if I dont check these and do

options := PRLIstOptions { Head: head, Base: base, State: "all"}

These should be empty by default?

Reply:
yes that's right, done